docs/content/config/environment.md

```diff
@@ -446,6 +446,12 @@ Can be used to control the score when the [`HFILTER_HOSTNAME_UNKNOWN` symbol](#r
 
 Default: 6 (which corresponds to the `add_header` action)
 
+##### RSPAMD_SPAM_SUBJECT
+
+This variable defines a prefix for e-mails tagged with the `X-Spam: Yes` header. In general, it works similar to [`SA_SPAM_SUBJECT`](#sa_spam_subject). However, the default is handled differently.
+
+Default: empty (no prefix will be added to e-mails)
```

Comment:
Big fan of changing this default 🙏

My main argument is that users can't edit subjects of received emails, and false-positives are annoying to reply/forward/etc.

Reply:
This was my initial instinct, and I agree with you, but we had a recent PR (I'll look it up) where we had a discussion on this default. I will need to take this into consideration.

I have an idea though: we should log a warning in case the user

1. did not set `MOVE_SPAM_TO_JUNK` **and**
2. did not set `RSPAMD_SPAM_SUBJECT`

in order to make everyone aware that spam might not be easily visible right away.

Reply:
From what I understand from our rspamd docs, a score of 11, is similar to triggering `SA_KILL` (_a score that identifies "definitely spam", or at least is meant to be fairly confident_), which we have a separate ENV `SPAMASSASSIN_SPAM_TO_INBOX` to prevent the action of reject/bounce/discard, and always deliver mail regardless to spam... but there is no equivalent support for rspamd integration right?

For SpamAssassin:
- We deliver all spam by default based on [this](https://github.com/docker-mailserver/docker-mailserver/issues/1396#issuecomment-597210154) reasoning.
- I also compiled [these findings for SA scores](https://github.com/docker-mailserver/docker-mailserver/pull/3058#issuecomment-1420268148), one had a [legitimate mail with a score of `8`](https://github.com/docker-mailserver/docker-mailserver/issues/768). There was also another request to always deliver by default (_since it'd go into the junk folder anyway_).

---

> It was argued that the prefix is useful, and I concur.

It's not that useful without explicit ENV changes?
- [We concluded that the value was primarily for POP3 users](https://github.com/docker-mailserver/docker-mailserver/pull/3767#discussion_r1456140100), no known IMAP users have an interest in it with the default to store spam in a separate junk folder. Perhaps if they used the `_SCORE_` token with a custom subject set, but again that is opt-in.
- We agreed that emitting a warning would be good `if IMAP=0 and POP3=1 and MOVE_SPAM_TO_JUNK=1` (_a spam prefix to a subject won't be valid here, hence the warning_).

Follow-up issues for changes related to that discussion:
- https://github.com/docker-mailserver/docker-mailserver/issues/3788
- https://github.com/docker-mailserver/docker-mailserver/issues/3791

---

- Make this feature anti-spam service agnostic if possible.
- Not enabled by default (breaking change)
- `undef` needs to be set for SA, and communicated that'd be no longer necessary to set.
- When it is configured, it should emit a warning log if `MOVE_SPAM_TO_JUNK=1` && `ENABLE_POP3=1` (_not sure if `ENABLE_IMAP=1` has relevance to preventing that sort of warning_).